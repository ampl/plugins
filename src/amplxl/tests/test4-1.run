reset;

set ND ;
set NS ;
set F ;
set F1 {ND} within F ;
set A1 within ((ND cross NS) union (NS cross (ND union NS))) ;

data;

set ND := DC1 DC2 ;
set NS := S1 S2 S3 S4 ; 

set A1 : DC1 DC2 S1 S2 S3 S4 :=
    DC1   -   -  +  +  +  +
    DC2   -   -  +  +  +  +
    S1    +   +  -  +  +  +
    S2    +   +  +  -  +  +
    S3    +   +  +  +  -  +
    S4    +   +  +  +  +  -    ;

set F := T1 T2 T3 T4 V1 V2 V3 V4 V5 V6 V7 V8; 
set F1[DC1] := T1 T2 ; 
set F1[DC2] := T3 T4 ;

model;
param T_LINKS_F1 {d in ND, t in F1[d], A1} >= 0 ;

### VERSION 1: ONE 1D TABLE

table T_LINKS_F1_tab IN "amplxl" "t_links-1.xlsx" "t_links_f1":
   [ND, F1, FROM, TO], T_LINKS_F1;

read table T_LINKS_F1_tab;

display T_LINKS_F1;

