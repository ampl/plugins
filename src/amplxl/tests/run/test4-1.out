T_LINKS_F1 [DC1,T1,*,*]
:     DC1     S1     S2     S3     S4     :=
DC1    .     0.55   0.43   0.62   0.57
S1    0.55    .     0.23   0.38   0.35
S2    0.43   0.23    .     0.3    0.35
S3    0.62   0.38   0.3     .     0.37
S4    0.57   0.35   0.35   0.37    .

 [DC1,T2,*,*]
:     DC1     S1     S2     S3     S4     :=
DC1    .     0.55   0.43   0.62   0.57
S1    0.55    .     0.23   0.38   0.35
S2    0.43   0.23    .     0.3    0.35
S3    0.62   0.38   0.3     .     0.37
S4    0.57   0.35   0.35   0.37    .

 [DC2,T3,*,*]
:     DC2     S1     S2     S3     S4     :=
DC2    .     0.62   0.62   0.45   0.57
S1    0.62    .     0.23   0.38   0.35
S2    0.62   0.23    .     0.3    0.35
S3    0.45   0.38   0.3     .     0.37
S4    0.57   0.35   0.35   0.37    .

 [DC2,T4,*,*]
:     DC2     S1     S2     S3     S4     :=
DC2    .     0.62   0.62   0.45   0.57
S1    0.62    .     0.23   0.38   0.35
S2    0.62   0.23    .     0.3    0.35
S3    0.45   0.38   0.3     .     0.37
S4    0.57   0.35   0.35   0.37    .
;

